File: practice/mathematics/number_theory/sherlock_and_gcd/sherlock_and_gcd.py
#!/usr/bin/env python

t = int(raw_input().strip())
for _ in xrange(t):
    n = int(raw_input().strip())
    a = map(int, raw_input().strip().split())
    
    m = min(a)
    if m == 1:
        print "YES"
    else:
        val = 2
        while val < m + 1:
            pos = 0
            while pos < n:
                if a[pos] % val != 0:
                    break
                pos += 1
            if pos == n:
                print "NO"
                break
            val += 1
        if val == m + 1:
            print "YES"

File: practice/mathematics/number_theory/sherlock_and_gcd/sherlock_and_gcd.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int t, n;
    
    cin >> t;
    while (t--)
    {
        cin >> n;
        vector<int> a(n);
        for (int pos = 0; pos < n; ++pos)
        {
            cin >> a[pos];
        }
        
        int m = *min_element(a.begin(), a.end());
        if (m == 1)
        {
            cout << "YES" << endl;   
        }
        else
        {
            int val = 2;
            while (val < m + 1)
            {
                int pos = 0;
                while (pos < n)
                {
                    if (a[pos] % val != 0)
                        break;
                    pos++;
                }
                if (pos == n)
                {
                    cout << "NO" << endl;
                    break;
                }
                val++;
            }
            if (val == m + 1)
            {
                cout << "YES" << endl;   
            }
        }
    }
    
    return 0;
}

